src/context/walletContext/walletContext.tsx

```diff
@@ -51,166 +43,137 @@ export const WalletProvider = ({ children }: { children: ReactNode }) => {
+    }, [selectedAddress, wallets, isWalletConnected])
 
-    /**
-     * Update selected wallet with connection status when kernel client is ready or wagmi is connected
-     */
+    // Set initial selected address
     useEffect(() => {
-        if (selectedWallet) {
-            updateSelectedWalletWithConnectionStatus(selectedWallet)
+        if (!selectedAddress && wallets.length > 0) {
+            const initialWallet = wallets.find(isPeanut) ?? wallets[0]
+            setSelectedAddress(initialWallet.address)
         }
-    }, [kernelClientAddress, wagmiAddress, selectedWallet])
-
-    /**
-     * Add a new BYOW to DB and local wallets prop.
-     * Called every time a new address is connected to the wagmi provider
-     * Will add new BYOW only if:
-     * - wagmiAddress is not undefined
-     * - wallets are not undefined and user logged in (PW available, at least)
-     */
+    }, [wallets, selectedAddress])
+
+    // Add new BYOW wallet when connected
     useEffect(() => {
-        if (!user) return
-        if (wagmiAddress && wallets && wallets.length > 0) {
-            // only check if user logged in (wallets will always include PW in this case) and wallets have been set up
-            const foundWallet = wallets.find((wallet: interfaces.IWallet) => {
-                wallet.address == wagmiAddress
+        if (!user || !wagmiAddress || !wallets.length) return
+
+        const walletExists = wallets.some((wallet) => wallet.address === wagmiAddress)
+        if (!walletExists) {
+            addAccount({
+                accountIdentifier: wagmiAddress,
+                accountType: interfaces.WalletProviderType.BYOW,
+                userId: user.user.userId as string,
             })
-            if (!foundWallet) {
-                // if currConnectedBYOW is not already in wallets
-                // addAccount() and then,
-                // it does NOT need to add a new wallet to the wallets prop,
-                // that is because addAccount() calls fetchUser() on success,
-                // and accounts are populated anew
-                addAccount({
-                    accountIdentifier: wagmiAddress,
-                    accountType: interfaces.WalletProviderType.BYOW,
-                    userId: user.user.userId as string, // will always be defined, since user logged in
-                })
-            }
         }
-    }, [wagmiAddress, wallets, user])
-
-    const removeBYOW = async () => {
-        // TODO: when successful API call, do NOT refetch all wallets (bad UX), but
-        // go on the wallet list and remove wallet
-        // TODO: if wallet connected to provider, disconnect wallet from provider
-        // then
-        // TODO: if active, default active -> PW if logged in, otherwise
-        // if another wallet is connected to the provider, make that active, otherwise
-        // no wallet active (have to review all props to ensure they are null)
+    }, [wagmiAddress, wallets, user, addAccount])
```

Comment:
Q: doesn't having addAccount in dep array make this circular?

Reply:
addAccount should not change, is an util function, it should not be part of this depArray